src/client/Microsoft.Identity.Client/Cache/SuggestedWebCacheKeyFactory.cs

```diff
@@ -54,8 +55,8 @@ private static bool GetOboOrAppKey(AuthenticationRequestParameters requestParame
 
             if (requestParameters.ApiId == TelemetryCore.Internal.Events.ApiEvent.ApiIds.AcquireTokenForClient)
             {
-                key = requestParameters.AppConfig.ClientId + "_AppTokenCache";
-                return true;
+                string tenantId = requestParameters.Authority.TenantId ?? "";
+                key = $"{requestParameters.AppConfig.ClientId}_{tenantId}_AppTokenCache";
```

Comment:
Test scenarios: 
- authority is "l.m.o/common" -> tenant id is common
- authority is tenanted -> tenant id
- authority does not support tenants (ADFS or B2C) -> ""

Reply:
Thanks @yohanb for proposing the fix!